### PR TITLE
Various docs improvements.

### PR DESCRIPTION
**Pull request by jojolepro**
Fixes https://github.com/amethyst/amethyst/issues/874 https://github.com/amethyst/amethyst/issues/870 https://github.com/amethyst/amethyst/issues/867

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/881)

### CODE REVIEW

**Comment by jojolepro** [comment]
Fixes https://github.com/amethyst/amethyst/issues/882

File: book/src/concepts/world.md [hunk 1]
```diff
@@ -115,6 +115,8 @@ All:
     world.delete_all().expect("Failed to delete all entities.");
 ```
 
+__Note: Entities are lazily deleted. They are effectively removed only at the end of each frame.__
```
Comment (Moxinilian) [inline book/src/concepts/world.md]: ``Note: Entities are lazily deleted, which means that deletion only happens at the end of the frame and not immediately when calling the `delete` method.``
Or something like that.

Comment (jojolepro) [reply book/src/concepts/world.md]: done

File: book/src/concepts/event-channel.md [hunk 2]
```diff
@@ -60,6 +60,9 @@ Then, when you want to read the events:
 Note that you only need to have a read access to the channel when reading events.
 It is the `ReaderId` that needs to be mutable to keep track of where your last read was.
 
+**IMPORTANT: The event channel automatically grows as events are added to it and shrinks only to a point where all readers have been through.
```
Comment (LucioFranco) [inline book/src/concepts/event-channel.md]: `and shrinks only to a point where all readers have been through.` I would change this to `and only decreases in size once all readers have read through the older events.`

File: book/src/concepts/world.md [hunk 3]
```diff
@@ -115,6 +115,8 @@ All:
     world.delete_all().expect("Failed to delete all entities.");
 ```
 
+__Note: Entities are lazily deleted, which means that deletion only happens at the end of the frame and not immediately when calling the `delete` method.__
```
Comment (LucioFranco) [inline book/src/concepts/world.md]: Just curious if the engine calls maintain here or if we want to kind of link this back to specs maintain? Just to complete the image of how specs integrates etc.

File: book/src/pong-tutorial/pong-tutorial-01.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ you're having trouble getting the project to run, double check the
 We've opened and closed a window, so we're basically done! But let's write this
 functionality ourselves so we're sure we know what's going on.
 
-In `src` there's a `main.rs` file. Delete everything, then add these imports:
+**In `src` there's a `main.rs` file. Delete everything, then add these imports:**
```
Comment (LucioFranco) [inline book/src/pong-tutorial/pong-tutorial-01.md]: It should be `Delete everything in that file`  to be more clear.

File: book/src/pong-tutorial/pong-tutorial-02.md [hunk 5]
```diff
@@ -28,14 +28,14 @@ Then, in `main.rs` declare a module:
 mod pong;
 ```
 
-And in the `run()` function add:
+And in main.rs, before `fn main()`, add this import:
```
Comment (LucioFranco) [inline book/src/pong-tutorial/pong-tutorial-02.md]: I would rephrase this as  `And in main.rs at the top of the file add this import.`

File: book/src/concepts/world.md (hunk 3 above)
Comment (jojolepro) [reply book/src/concepts/world.md]: yes, but I just removed it per request from a previous review

Comment (LucioFranco) [reply book/src/concepts/world.md]: I still see it in the latest commit?

Comment (jojolepro) [reply book/src/concepts/world.md]: I'm sure I pushed it. https://github.com/amethyst/amethyst/pull/881/commits/984b3dbdce56d860d115cd1437b7be3c660ad5c1

**Comment by jojolepro** [comment]
Let's see if bors is happy with my changes :crossed_fingers: 
bors r+

we love you bors. pray for bors.

**Comment by jojolepro** [comment]
bors r-

gotta squash

**Comment by stenverbois** [comment]
This would be a good PR to remove all those `ignore`s on code snippets. That way you won't have to worry about them being out of date as they are checked by the compiler.

**Comment by jojolepro** [comment]
except the only one I did that with I was unable to get it to compile at all. Definitely something to do if you know how though.

**Comment by LucioFranco** [comment]
@jojolepro we should open an issue for this as its something that someone getting into amethyst could work on.

File: book/src/game-data.md [hunk 6]
```diff
@@ -81,7 +81,8 @@ impl<'a, 'b> CustomGameDataBuilder<'a, 'b> {
 
 impl<'a, 'b> DataInit<CustomGameData<'a, 'b>> for CustomGameDataBuilder<'a, 'b> {
     fn build(self, world: &mut World) -> CustomGameData<'a, 'b> {
-        let pool = world.read_resource::<ThreadPool>().clone();
+        // Get a handle to the `ThreadPool`.
+        let pool = world.read_resource::<Arc<ThreadPool>>().clone();
```
Comment (VermillionAzure) [inline book/src/game-data.md]: Why did we change to `Arc`?

File: book/src/pong-tutorial/pong-tutorial-02.md [hunk 7]
```diff
@@ -394,10 +394,9 @@ As it turns out, the system we're missing is `TransformSystem`, and we can add i
 `TransformBundle`.
 
 ```rust,ignore
-let mut game = Application::build("./", Pong)?
-    .with_bundle(TransformBundle::new())? // Add this bundle
+let game_data = GameDataBuilder::default()
```
Comment (VermillionAzure) [inline book/src/pong-tutorial/pong-tutorial-02.md]: Outsider question: is the builder idiom still idiomatic for Rust?

**Comment by Moxinilian** [comment]
At some point I will use the `#` utility that allows to provide context to code snippet so that they can be built even without showing all the boilerplate.

**Comment by Moxinilian** [comment]
Actually maybe right now.

File: book/src/pong-tutorial/pong-tutorial-02.md (hunk 7 above)
Comment (jojolepro) [reply book/src/pong-tutorial/pong-tutorial-02.md]: yes

File: book/src/game-data.md (hunk 6 above)
Comment (jojolepro) [reply book/src/game-data.md]: because we use a Arc<ThreadPool> and the example was wrong.

**Comment by LucioFranco** [comment]
@jojolepro i think it timed out due to appveyor.
